contracts/nft/nft.ral

```diff
@@ -5,11 +5,19 @@ Contract NFT(
   tokenUri: ByteVec,
   nftIndex: U256
 ) implements INFT {
+  enum ErrorCodes {
+    WrongCollection = 0
+  }
+
   pub fn getTokenUri() -> ByteVec {
     return tokenUri
   }
 
+  // We do not need this after introducing parentId instr
+  @using(checkExternalCaller = false)
   pub fn getCollectionId() -> ByteVec {
+    let expectedTokenContract = INFTCollection(collectionId).nftByIndex(nftIndex)
+    assert!(selfContractId!() == contractId!(expectedTokenContract), ErrorCodes.WrongCollection)
     return collectionId
```

Comment:
Since `getCollectionId` is not part of the std, do we need to remove `collectionId` from the implementation?

Reply:
Okey, i will remove. Lets pass in the `collectionId`